### ISSUE TITLE
feat(common): add spell checker

### ISSUE THREAD

**Issue by JaeAeich** [elixir-cloud-aai/actions#4]
Add spell checker  as common action instead of language specific so we can rely on only one dep that can be used to cross language.

**Comment by uniqueg** [comment]
Not really important for now, I guess, but perhaps it would be good to have a common spellchecker method and dictionary, but the ability to extend with custom dictionary for each repo.

**Comment by JaeAeich** [comment]
yeah, my point for this was that the current setup for actions, I have been taking reference to, is from cookicutter, so while writing CI for python, I only configured spell checker in python, which didn't make sense (as I was simultaneously wrting CI for rust), so I thought why not use the same dep (be it python, rust, node, or even C) for all the repo as this action should be language agnostic.

So the current changes that I am making is, for poetry setup action which used to have fields like `install-options` etc to take in groups from `pyporject` to install, I am planning on adding feat to install independent dep so that when we can install the same dep (say typos that we use for python) in rust projs as well.
